sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala

```diff
@@ -57,19 +57,21 @@ object ResolveTableValuedFunctions extends Rule[LogicalPlan] {
    * A TVF maps argument lists to resolver functions that accept those arguments. Using a map
    * here allows for function overloading.
    */
-  private type TVF = Map[ArgumentList, Seq[Any] => LogicalPlan]
+  private type TVF = Map[ArgumentList, (UnresolvedTableValuedFunction, Seq[Any]) => LogicalPlan]
 
   /**
    * TVF builder.
    */
-  private def tvf(args: (String, DataType)*)(pf: PartialFunction[Seq[Any], LogicalPlan])
-      : (ArgumentList, Seq[Any] => LogicalPlan) = {
+  private def tvf(args: (String, DataType)*)(
+      pf: PartialFunction[(UnresolvedTableValuedFunction, Seq[Any]), LogicalPlan])
```

Comment:
Hmmm... this signature is kind of complex. Can we try to use some kind of class/case class that encapsulates this?